index.tsx

```diff
@@ -982,6 +1017,11 @@ export function compiler(
       )
     )
 
+    arr = arr.filter(item => item)
```

Comment:
This probably isn't a safe transform to perform since `0` is falsy. I think we can detect nullish output in `reactFor` and filter out crud at that point to avoid needing a full traversal later.

Reply:
This change in `reactFor` should do it:

```diff
  if (isString && lastWasString) {
    result[result.length - 1] += nodeOut
- } else { 
+ } else if (nodeOut !== null) {
     result.push(nodeOut)
  }
```

Reply:
Let's try this.